src/chocolatey.tests/infrastructure.app/commands/ChocolateyCacheCommandSpecs.cs

```diff
@@ -0,0 +1,126 @@
+            protected ChocolateyConfiguration Configuration = new ChocolateyConfiguration();
+            protected Mock<IFileSystem> FileSystem = new Mock<IFileSystem>();
+
+            public override void Context()
+            {
+                Configuration.CommandName = "cache";
+                Command = new ChocolateyCacheCommand(FileSystem.Object);
+            }
+        }
+
+        public class WhenImplementingCommandFor : ChocolateyCacheCommandSpecsBase
+        {
+            private List<CommandForAttribute> _results;
+
+            public override void Because()
+            {
+                _results = Command.GetType().GetCustomAttributes<CommandForAttribute>().ToList();
+            }
+
+            [Fact]
+            public void ShouldImplementCache()
+            {
+                _results.Should().AllSatisfy(c => c.CommandName.Should().Be("cache"));
+            }
+
+            [Fact]
+            public void ShouldSetADescription()
+            {
+                _results.Should().AllSatisfy(c => c.Description.Should().NotBeNullOrEmpty());
+            }
+
+            [Fact]
+            public void ShouldSetVersionProperty()
+            {
+                _results.Should().AllSatisfy(c => c.Version.Should().Be("2.1.0"));
+            }
+        }
+
+        public class WhenConfiguringTheArgumentParser : ChocolateyCacheCommandSpecsBase
+        {
+            private OptionSet _optionSet;
+
+            public override void Context()
+            {
+                base.Context();
+                _optionSet = new OptionSet();
+            }
+
+            public override void Because()
+            {
+                Command.ConfigureArgumentParser(_optionSet, Configuration);
+            }
+
+            [Fact]
+            [InlineData("expired")]
+            public void ShouldAddOptionToOptionSet(string name)
+            {
+                _optionSet.Contains(name).Should().BeTrue("Option set should include the parameter {0}", name);
+            }
+        }
+
+        public class WhenParsingAdditionalParameters : ChocolateyCacheCommandSpecsBase
+        {
+            public override void Because()
+            {
+                
+            }
+
+            public override void BeforeEachSpec()
+            {
+                Configuration.CacheCommand.Command = CacheCommandType.Unknown;
+                MockLogger.Reset();
+            }
+
+            [Fact]
+            public void ShouldHaveSetCacheCommandTypeToListOnUnusedSubCommand()
+            {
+                Command.ParseAdditionalArguments(new List<string>(), Configuration);
+                Configuration.CacheCommand.Command.Should().Be(CacheCommandType.List);
+            }
+
+            [Fact]
+            public void ShouldHaveSetCacheCommandTypeToListOnListSubCommand()
+            {
+                var unparsedArgs = new[] { "list" };
+                Command.ParseAdditionalArguments(unparsedArgs, Configuration);

```

Comment:
I think we are missing an assertion on this test, right?  Similar to line 100 above.

Reply:
THis has been added.